lib/KTable/index.vue

```diff
@@ -294,156 +294,144 @@
+            cellAndFocusableElements[focusedIndex + 1].focus();
+            event.preventDefault();
+            return;
+          } else {
+            // Move to the first focusable element of the next cell
+            if (colIndex < totalCols - 1) {
+              nextColIndex = colIndex + 1;
+            } else if (rowIndex < totalRows - 1) {
+              nextColIndex = 0;
+              nextRowIndex = rowIndex + 1;
             } else {
-              if (!event.shiftKey) {
-                if (colIndex < totalCols - 1) {
-                  nextColIndex = colIndex + 1;
-                } else if (rowIndex < totalRows - 1) {
-                  nextColIndex = 0;
-                  nextRowIndex = rowIndex + 1;
-                } else {
-                  // Allow default behavior when reaching the last cell
-                  return;
-                }
-              } else {
-                if (colIndex > 0) {
-                  nextColIndex = colIndex - 1;
-                } else if (rowIndex > 0) {
-                  nextColIndex = totalCols - 1;
-                  nextRowIndex = rowIndex - 1;
-                } else {
-                  // Allow default behavior when reaching the first cell
-                  return;
-                }
-              }
+              return; // Allow default Tab behavior when reaching the end
             }
-
-            break;
+            const nextCell = this.getCell(nextRowIndex, nextColIndex);
+            const nextFocusableElements = this.getFocusableElements(nextCell);
+            const nextCellAndFocusableElements = [nextCell, ...nextFocusableElements];
+            nextCellAndFocusableElements[0].focus();
+            this.updateFocusState(nextRowIndex, nextColIndex);
+            event.preventDefault();
           }
-
-          default:
+        } else {
+          // Shift+Tab key navigation
+          if (focusedIndex < cellAndFocusableElements.length - 1) {
+            // Move to the previous focusable element within the cell
+            cellAndFocusableElements[focusedIndex + 1].focus();
+            event.preventDefault();
             return;
+          } else {
+            // Move to the last focusable element of the previous cell
+            if (colIndex > 0) {
+              nextColIndex = colIndex - 1;
+            } else if (rowIndex > 0) {
+              nextColIndex = totalCols - 1;
+              nextRowIndex = rowIndex - 1;
+            } else {
+              return; // Allow default Shift+Tab behavior when at the beginning
+            }
+            const prevCell = this.getCell(nextRowIndex, nextColIndex);
+            const prevFocusableElements = this.getFocusableElements(prevCell);
+            const prevCellAndFocusableElements = [prevCell, ...prevFocusableElements];
+            prevCellAndFocusableElements[prevCellAndFocusableElements.length - 1].focus();
+            this.updateFocusState(nextRowIndex, nextColIndex);
```

Comment:
This "update focus state" is already taking care of the cell focus, and we are just focusing the prevCell, and not taking into account the "prevCellAndFocusableElements" anyways. We should keep this method, without trying to focus the last focusable element of the previous cell. That means, removing these lines:

```javascript
            const prevCell = this.getCell(nextRowIndex, nextColIndex);
            const prevFocusableElements = this.getFocusableElements(prevCell);
            const prevCellAndFocusableElements = [prevCell, ...prevFocusableElements];
            prevCellAndFocusableElements[prevCellAndFocusableElements.length - 1].focus();
```